setup.py

```diff
@@ -102,7 +102,47 @@ def run(self):
                     shutil.rmtree(os.path.join(dirpath, dirname))
 
 
-cmdclass = {'clean': CleanCommand}
+def get_openmp_flag(compiler):
+    if sys.platform == "win32" and ('icc' in compiler or 'icl' in compiler):
+        return ['/Qopenmp']
+    elif sys.platform == "win32":
+        return ['/openmp']
+    elif sys.platform == "darwin" and ('icc' in compiler or 'icl' in compiler):
+        return ['-openmp']
+    elif sys.platform == "darwin" and 'openmp' in os.getenv('CC', ''):
+        # -fopenmp can't be passed as compile arg when using apple clang
+        return ['']
+    return ['-fopenmp']
```

Comment:
May I suggest the following:

```python
def get_openmp_flag(compiler):
    if 'openmp' in os.getenv('CC', '') or 'openmp' in os.getenv('CXX', ''):
        # Respect the flags manually passed by the user as an environment
        # variable.
        # In particular, our macOS wheel build jobs use the following environment
        # variables to build with Apple clang and the brew installed "libomp":
        # 
        # export CC="clang -Xpreprocessor -fopenmp"
        # export CFLAGS="$CFLAGS -I/usr/local/opt/libomp/include"
        # export CXX="clang++ -Xpreprocessor -fopenmp"
        # export CXXFLAGS="$CXXFLAGS -I/usr/local/opt/libomp/include"
        # export LDFLAGS="$LDFLAGS -L/usr/local/opt/libomp/lib -lomp"
        # export DYLD_LIBRARY_PATH=/usr/local/opt/libomp/lib
        return ['']
    if sys.platform == "win32" and ('icc' in compiler or 'icl' in compiler):
        return ['/Qopenmp']
    elif sys.platform == "win32":
        return ['/openmp']
    elif sys.platform == "darwin" and ('icc' in compiler or 'icl' in compiler):
        return ['-openmp']
    # Default flag for GCC and clang:
    return ['-fopenmp']
```

Reply:
Note that I am not 100% sure if the above comment is correct for C++ (CXX and CXXFLAGS).

Reply:
A comment can't hurt. However the reason is different and only apply to mac.
Normally `-fopenmp` is a compile flag (and also a link flag), but on default apple clang, it's not supported and therefore should not appear in the compile command. However, it is supported as a preprocessor flag, hence `-Xpreprocessor -fopenmp`. I'll modify the comment to explain that.

Also, I'm not sure setting the `-fopenmp` flag in CC would work because it should be a compile flag. So I'm in favor to keep the CC check only on darwin platform. 

Putting it in CFLAGS would work for sure, but I don't override the CFLAGS var (I just append to it) so if someone sets the flags manually it'll work, so I don't think there's a need to check the CFLAGS.

Reply:
Alright.

Reply:
Then why not setup that as a processor flag explicitly with the `CPPFLAGS` variable instead of using the `CC` variable:

export CPPFLAGS="$CPPFLAGS -Xpreprocessor -fopenmp"

Reply:
Ok seems to work.